### DISCUSSION TITLE
Upscale images from directory

### DISCUSSION THREAD

**Discussion by Ariellapin** [lllyasviel/Fooocus#2174] [General]
I'd like to propose a new feature for the Stable Diffusion Web UI: batch upscaling images from a directory. This feature would allow users to easily upscale multiple images at once, saving time and effort compared to the current process of upscaling each image individually.

**Comment by mashb1t** [comment]
See https://github.com/lllyasviel/Fooocus/issues/1236

This is 100% doable, but quality will be worse than using manual upscaling.
Would you propose only latent (fast) upscaling or AI upscaling with SD (basically hires fix)?
If you mean the latter, one will also have to provide a prompt, which may be autogenerated using describe.
Did you think about this and what is your opinion?

**Reply by Ariellapin** [reply]
Thank you for the insightful points! I agree that quality could be impacted with batch upscaling, especially compared to manual processing. However, the convenience and efficiency benefits would still be significant for many users.
Upscaling with SD :
1.Autogenerated Prompts: The UI could analyze images and attempt to generate basic prompts using "describe" functionality. This might work for simple images but could falter for complex scenes.
2.Text File with File Name and Prompt: This allows users to provide custom prompts tailored to each image in a text file alongside filenames. This offers more control and potentially better results but requires user effort.

Imagine this: You upload a bunch of images and their corresponding prompts in a text file to upscale . As the UI processes each image, magic happens! 

If the prompt perfectly matches the image content (think "portrait of a cat" for a picture of a fluffy feline), a little green checkmark icon appears next to the filename. This instantly tells you, "Hey, you nailed it with this prompt!"
If there's a good but not perfect match (like "a majestic animal" for the same cat picture), a yellow lightbulb icon pops up. This suggests, "Hmm, this prompt captures the essence, but maybe try something more specific?"
And for filenames without matching prompts, a neutral gray icon remains. This signals, "Time to unleash your creativity and craft the perfect prompt for this image!"
This visual feedback system makes the upscaling process intuitive and efficient. You can instantly see which prompts work well and focus on refining those that need a little nudge. Plus, it adds a touch of fun and gamification to the experience!

**Reply by mashb1t** [reply]
This sounds like an interesting idea, but i personally don't have the time to implement it and i doubt that the owner of Fooocus is going to respond anytime soon :/
Still waiting for some insights for dev roadmap: https://github.com/lllyasviel/Fooocus/discussions/2154

**Reply by patrickchamp** [reply]
I thought I'd add a suggestion here instead of submitting a new idea. It would be useful if maybe having "Input Image > Upscale" open without an input image selected that it would make it perform an upscale with each image generation. Not as ideal as a batch folder option but at least this wouldn't complicate the UI by adding anything other than some explanation text and it could make use of the current prompt (and current wildcard) as you let it run a batch.

**Comment by Ariellapin** [comment]
Unfortunately, many stable diffusion models do not produce satisfactory results, especially when the person is standing far away and their face is unclear. Thus, upscaling is the best method to obtain a very good quality image. Ideally, I would write a feature that scales images and compresses them to produce a smaller size and better quality.
